apps/api/src/app/auth/services/passport/jwt.strategy.ts

```diff
@@ -1,25 +1,35 @@
+import type http from 'http';
 import { ExtractJwt, Strategy } from 'passport-jwt';
 import { PassportStrategy } from '@nestjs/passport';
 import { Injectable, UnauthorizedException } from '@nestjs/common';
-import { UserSessionData } from '@novu/shared';
+import { ApiAuthSchemeEnum, UserSessionData } from '@novu/shared';
 import { AuthService, Instrument } from '@novu/application-generic';
+import { HttpRequestHeaderKeysEnum } from '../../../shared/framework/types';
 
 @Injectable()
 export class JwtStrategy extends PassportStrategy(Strategy) {
   constructor(private readonly authService: AuthService) {
     super({
       jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
       secretOrKey: process.env.JWT_SECRET,
+      passReqToCallback: true,
     });
   }
 
   @Instrument()
-  async validate(payload: UserSessionData) {
-    const user = await this.authService.validateUser(payload);
+  async validate(req: http.IncomingMessage, session: UserSessionData) {
+    const user = await this.authService.validateUser(session);
     if (!user) {
       throw new UnauthorizedException();
     }
 
-    return payload;
+    session.scheme = ApiAuthSchemeEnum.BEARER;
```

Comment:
The scheme property is used to detect if the API is accessed from JWT or APIKey.